hw/mcu/nordic/nrf52xxx/src/hal_watchdog.c

```diff
@@ -44,13 +44,20 @@ nrf52_wdt_irq_handler(void)
 int
 hal_watchdog_init(uint32_t expire_msecs)
 {
-    uint64_t expiration;
-
     NRF_WDT->CONFIG = WDT_CONFIG_SLEEP_Msk;
 
-    /* Convert msec timeout to counts of a 32768 crystal */
-    expiration = ((uint64_t)expire_msecs * 32768) / 1000;
-    NRF_WDT->CRV = (uint32_t)expiration;
+    if (expire_msecs >= 1048576) {
+        /* watchdog with more than 17 minutes? */
+        assert(0);
+    } else {
+        uint32_t e = expire_msecs, rtc = 32768, d = 1000;
```

Comment:
This is definitely not mynewt coding style: declaring multiple local variables in one line and initializing them all on the same line, too.